src/components/ChatEntry.js

```diff
@@ -1,22 +1,36 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
-const ChatEntry = (props) => {
+const ChatEntry = ({id, sender, body, timeStamp, liked, type, color, onLike}) => {
+  const entryClass = type === 'local' ? 'local' : 'remote';
+  const bubbleClass = type === 'local' ? 'local' : 'remote';
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+    <div className={`chat-entry ${entryClass}`}>
+      <h2 className={`entry-name ${bubbleClass}`}style={{ color: color }}>
```

Comment:
We should avoid inline styles to separate the layer of or web app and keep everything organized. Most modern browsers support CSS custom properties, which we can use to achieve a similar effect. (Info on CSS Custom Properties support: https://caniuse.com/css-variables). 

What we could do is declare a css variable that is set to the default color of black, and use that variable in a css rule that applies to the element we want to see the change on. Then when a new color is selected, we would update that CSS variable.
```css
// In ChatEntry.css

// Create a CSS custom property (variable) that's globally accessible
// we could use an element other than root, but we would need to look up 
// the --curr-color var on that element when we want to change it later.
:root {
  --curr-color: black;
}

// Declare a rule that targets the sender names and uses the CSS variable
.chat-entry h2 {
  color: var(--curr-color);
}
```  

```js
// In ColorChoice.js

const handleColorChange = (e) => {
    // Update the state for the color picker
    const selectedColor = e.target.value;
    setColor(selectedColor);

    // Update the CSS variable to the chosen color
    // This is a little hacky, we would probably still have a piece of react state in App.js that we update here
    // but in App.js, somewhere before returning the JSX we would call the line below to ensure the CSS
    // var matches the React state, ensuring the React State is still our source of truth for what the user sees
    document.documentElement.style.setProperty("--curr-color", selectedColor);
};
```

Reply:
Here's a link to an article that dives into what I'm talking about above: https://spacejelly.dev/posts/how-to-create-css-custom-properties-that-dynamically-update-with-react-javascript/